apioak/plugin/key-auth.lua

```diff
@@ -34,26 +37,27 @@ function _M.http_access(oak_ctx)
     if not oak_ctx.plugins[_M.key] then
         return false, nil
     end
-    --get key
-    local key = pdk.request.header('apikey')
-    if not key then
-        pdk.response.exit(401, { err_message = "Missing API key found in request" })
+
+    local plugin_conf = oak_ctx.plugins[_M.key]
+
+    ngx.say(pdk.json.encode(plugin_conf));
+
+    local _, err = pdk.schema.check(schema, plugin_conf)
+    if err then
+        return false, nil
     end
 
-    --get etcd keys
-    local etcd_cli = pdk.etcd.new()
-    local keys_etcd = etcd_cli.get(_M.name)
-    local keys_body = keys_etcd.body
-    if (not keys_body.node) or (not keys_body.node.value) then
-        pdk.response.exit(401, { err_message = "API key no set" })
+    local header_key = pdk.request.header('apikey')
+    if not header_key then
```

Comment:
The key to obtain the authentication value needs to use `Authorization` instead of `apikey`.

Reply:
The meaning of the `header_key` variable is not clear. It is recommended to use `secret_key`.